agent/tests/LoopFake.php

```diff
@@ -210,6 +213,10 @@ public function run(): void
                     'periodic' => $pendingTimer['periodic'],
                 ], $this->pendingTimers);
 
+                foreach ($this->writeStreams as [$stream, $listener]) {
+                    $listener($stream);
+                }
+
```

Comment:
I've scattered these everywhere as a temporary measure to get this moving.

Need to return to this on Monday to get it happening in one place.